tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java

```diff
@@ -737,9 +771,15 @@ private void mediaLoadCompleted(final ResultElement<PKMediaEntry> response, fina
 
         final PKMediaEntry entry = response.getResponse();
         if (entry != null) {
+            if (entriesCache != null && initOptions.mediaEntryLruCacheConfig.getAllowMediaEntryCaching() && entry.getMetadata() != null && entry.getMetadata().get("mediaAssetUUID") != null) {
+                log.d("Add Entry to Cache: name = " + entry.getName() + " mediaId = " + entry.getId());
+                String mediaEntryJson = new Gson().toJson(entry);
+                entriesCache.put(entry.getMetadata().get("mediaAssetUUID"), mediaEntryJson);
```

Comment:
where is `mediaAssetUUID` coming from?

Reply:
I set it on the PhoenixMediaProvider in the metadata

Reply:
done